test/core/client_channel/http_proxy_mapper_test.cc

```diff
@@ -32,12 +36,55 @@
 #include "test/core/util/scoped_env_var.h"
 #include "test/core/util/test_config.h"
 
+// Need to be in the same namespace as absl::optional...
+namespace absl {
```

Comment:
It's not okay to put code into the `absl` namespace, since we don't own that namespace.

Instead, I suggest you change the matcher to explictly print whatever you want.  For example, you can replace line 85 with the following:

```
if (*address_string != address) {
  *result_listener << "Actual address: " << *address_string;
  return false;
}
return true;
```

Alternatively, you can explicitly print things in the individual tests via something like this:

```
EXPECT_THAT(address, AddressEq("1.2.3.4"))
    << grpc_sockaddr_to_string(address, true).value_or("unknown");
```

Reply:
I removed these function, but I would like to suggest considering building a library of functions as it would be a significant boost to productivity. E.g. we could consider moving `AddressEq` matcher to a shared header.

Reply:
I'd be fine with having a common library of matchers.  We already have a similar one here: https://github.com/grpc/grpc/blob/a0c1027bb3da36bfa5d3eb2630fd62dd79d41608/test/core/resolver/endpoint_addresses_test.cc#L48

Reply:
Ack. Will do some code search in our code base at a later point and see if we can consolidate/reuse matchers.